src/components/graph/network-modification-tree-model.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { getLayoutedElements } from './layout';
+
+export default class NetworkModificationTreeModel {
+    treeElements = [];
+
+    updateLayout() {
+        this.treeElements = getLayoutedElements(this.treeElements);
+    }
+
+    addChild(element, parentId) {
+        // Add node
+        this.treeElements.push({
+            id: element.id,
+            type: element.type, // input node
+            data: { label: element.name, description: element.description },
+        });
+        // Add edge between node and its parent
+        this.treeElements.push({
+            id: 'e' + parentId + '-' + element.id,
+            source: parentId,
+            target: element.id,
+            type: 'smoothstep',
+        });
+        if (element.children) {
+            element.children.forEach((child) => {
+                this.addChild(child, element.id);
+            });
+        }
+    }
+
+    removeNodes(deletedNodes) {
```

Comment:
this function need comments, it's not trivial

Reply:
Added some comments

Reply:
thanks